pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -260,40 +260,35 @@ func reportMetrics(pa *autoscalingv1alpha1.PodAutoscaler, pc podCounts) {
 //	| -1   | >= min | >0    | active     | active     |
 func computeActiveCondition(ctx context.Context, pa *autoscalingv1alpha1.PodAutoscaler, pc podCounts) {
 	minReady := activeThreshold(ctx, pa)
+
 	if pc.ready >= minReady && pa.Status.ServiceName != "" {
 		pa.Status.MarkScaleTargetInitialized()
 	}
 
 	switch {
-	// Need to check for minReady = 0 because in the initialScale 0 case, pc.want will be -1.
-	case pc.want == 0 || minReady == 0:
-		if pa.Status.IsActivating() && minReady > 0 {
-			// We only ever scale to zero while activating if we fail to activate within the progress deadline.
-			pa.Status.MarkInactive("TimedOut", "The target could not be activated.")
-		} else {
-			pa.Status.MarkInactive(noTrafficReason, "The target is not receiving traffic.")
-		}
+	case pa.Status.IsActivating() && pc.want == 0 && minReady > 0:
+		pa.Status.MarkInactive("TimedOut", "The target could not be activated.")
 
-	case pc.ready < minReady:
-		if pc.want > 0 || !pa.Status.IsInactive() {
-			pa.Status.MarkActivating(
-				"Queued", "Requests to the target are being buffered as resources are provisioned.")
-		} else {
-			// This is for the initialScale 0 case. In the first iteration, minReady is 0,
-			// but for the following iterations, minReady is 1. pc.want will continue being
-			// -1 until we start receiving metrics, so we will end up here.
-			// Even though PA is already been marked as inactive in the first iteration, we
-			// still need to set it again. Otherwise reconciliation will fail with NewObservedGenFailure
-			// because we cannot go through one iteration of reconciliation without setting
-			// some status.
-			pa.Status.MarkInactive(noTrafficReason, "The target is not receiving traffic.")
-		}
+	case !pa.Status.IsInactive() && (pc.want == 0 || minReady == 0):
+		pa.Status.MarkInactive(noTrafficReason, "The target is not receiving traffic.")
+
+	case !pa.Status.IsInactive() && pa.IsUnreachable() && pc.ready < minReady:
+		pa.Status.MarkInactive("Failed", "The target is unreachable.")
+
+	case (!pa.Status.IsInactive() || pc.want > 0) && pc.ready < minReady:
+		pa.Status.MarkActivating("Queued", "Requests to the target are being buffered as resources are provisioned.")
 
 	case pc.ready >= minReady:
+
 		if pc.want > 0 || !pa.Status.IsInactive() {
 			pa.Status.MarkActive()
 		}
+
+	default:
+		cond := pa.Status.GetCondition("Active")
+		pa.Status.MarkInactive(cond.Reason, cond.Message)
```

Comment:
This doesn't seem right - given the reason and message wouldn't make sense when the condition is Active=False